python/CSXCAD/CSProperties.pyx

```diff
@@ -452,25 +467,39 @@ cdef class CSPropMaterial(CSProperties):
         """
         return (<_CSPropMaterial*>self.thisptr).GetIsotropy()
 
-    def SetMaterialProperty(self, **kw):
-        """ SetMaterialProperty(**kw)
-        Set the material properties.
-
-        :params epsilon: scalar or vector - relative electric permittivity
-        :params mue:     scalar or vector - relative magnetic permeability
-        :params kappa:   scalar or vector - electric conductivity
-        :params sigma:   scalar or vector - magnetic conductivity
-        :params density: float            - Density
-        """
-        for prop_name in kw:
-            val = kw[prop_name]
+    def SetMaterialProperty(self, epsilon:float|list=None, mue:float|list=None, kappa:float|list=None, sigma:float|list=None, density:float=None):
+        """Set the material properties.
+
+        :params epsilon: scalar or vector - Relative electric permittivity.
+        :params mue:     scalar or vector - Relative magnetic permeability.
+        :params kappa:   scalar or vector - Electric conductivity.
+        :params sigma:   scalar or vector - Magnetic conductivity.
+        :params density: float            - Density.
+        """
+        # Validate epsilon, mue, kappa, sigma:
+        for name,val in dict(epsilon=epsilon, mue=mue, kappa=kappa, sigma=sigma).items():
+            if val is not None:
+                if isinstance(val, (int,float)):
+                    continue
+                elif  isinstance(val, (tuple,list)) and len(val)==3 and all([isinstance(_, (int,float)) for _ in val]):
+                    continue
+                else:
+                    raise ValueError(f'`{name}` must be a scalar or a 3-vector of scalars, received `{val}`. ')
+        # Validate density:
+        if density is not None:
+            if not isinstance(density, (int, float)):
+                raise TypeError(f'`density` must be a float number, received object of type {type(density)}. ')
+        # Do the same as in the previous implementation:
+        kw = dict(epsilon=epsilon, mue=mue, kappa=kappa, sigma=sigma, density=density)
+        for prop_name,val in kw.items():
+            if val is None:
+                continue
             if prop_name == 'density':
                 (<_CSPropMaterial*>self.thisptr).SetDensity(val)
                 continue
-            if type(val)==float or type(val)==int:
+            if isinstance(val, (int,float)):
```

Comment:
It is better as it was but again a bit restrictive?  We should simply test it is an iterable and if not try to convert to float instead?

Reply:
Isn't it the same as before my change, but only written more compact? In both cases you make sure `val` is either `float` or `int`.